### PR TITLE
Contributing lib/ansible/modules/network/cloudengine/ce_vrf.py module to manage HUAWEI data center CloudEngine

### PR DESCRIPTION
**Pull request by QijunPan**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ce_vrf.py
##### ANSIBLE VERSION

```
ansible 2.3.0
config file = /etc/ansible/ansible.cfg
configured module search path = ['/home/ansible-2.1.1.0/lib/ansible/modules/core/network/cloudengine', '/home/ansible-2.1.1.0/lib/ansible/module_utils', '/home/ansible-2.1.1.0/lib/ansible/modules', '/home/ansible-2.1.1.0', '/home/ansible-2.1.1.0/lib/ansible/utils/']
```

##### SUMMARY

```
TASK [Config a vpn named vpna, description is test] ****************************
task path: /usr1/code/OpenNESS/code/current/KVM/intg/test/testcases/test-ce_vrf.yml:15
Using module file /usr/local/lib/python2.7/site-packages/ansible-2.3.0-py2.7.egg/ansible/modules/core/network/cloudengine/ce_vrf.py
<ce_6850> ESTABLISH LOCAL CONNECTION FOR USER: root
<ce_6850> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1487054159.22-62399001042523 `" && echo ansible-tmp-1487054159.22-62399001042523="` echo $HOME/.ansible/tmp/ansible-tmp-1487054159.22-62399001042523 `" ) && sleep 0'
<ce_6850> PUT /tmp/tmpDHp21I TO /root/.ansible/tmp/ansible-tmp-1487054159.22-62399001042523/ce_vrf.py
<ce_6850> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1487054159.22-62399001042523/ /root/.ansible/tmp/ansible-tmp-1487054159.22-62399001042523/ce_vrf.py && sleep 0'
<ce_6850> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1487054159.22-62399001042523/ce_vrf.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1487054159.22-62399001042523/" > /dev/null 2>&1 && sleep 0'
changed: [ce_6850] => {
    "changed": true, 
    "end_state": {
        "description": "test", 
        "vrf": "vpna"
    }, 
    "existing": {}, 
    "invocation": {
        "module_args": {
            "auth_pass": null, 
            "authorize": false, 
            "description": "test", 
            "host": "ce_6850", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 12345, 
            "provider": null, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": 10, 
            "transport": null, 
            "use_ssl": false, 
            "username": "rootDC", 
            "validate_certs": true, 
            "vrf": "vpna"
        }, 
        "module_name": "ce_vrf"
    }, 
    "proposed": {
        "description": "test", 
        "state": "present", 
        "vrf": "vpna"
    }, 
    "updates": [
        "ip vpn-instance vpna", 
        "description test"
    ]
}
```

### CODE REVIEW

File: lib/ansible/modules/network/cloudengine/ce_vrf.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+    - Manages VPN instance of HUAWEI CloudEngine switches.
+author: Yang yang (@CloudEngine-Ansible)
+notes:
+    - If no vrf is supplied, vrf is set to default.
+      If state==absent, the route will be removed, regardless of the
+      non-required parameters.
+options:
+    vrf:
+        description:
+            - VPN instance,the length of vrf name is 1 - 31,i.e. "test",but can not be _public_.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vrf.py]: `VPN instance, the length of vrf name is 1 - 31, i.e. "test",but can not be C(_public_).`

File: lib/ansible/modules/network/cloudengine/ce_vrf.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+        required: true
+    description:
+        description:
+            - Description of the vrf,the string length is 1 - 242 .
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vrf.py]: Missing space between `vrf,` and `the`.

File: lib/ansible/modules/network/cloudengine/ce_vrf.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+                    else:
+                        return True
+            return self.state == "present"
+        else:
+            return self.state == "present"
+
+    def check_params(self):
+        """Check all input params"""
+        # vrf check
+        if not self.vrf:
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vrf.py]: Not needed as you have `required=True`

File: lib/ansible/modules/network/cloudengine/ce_vrf.py [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+DOCUMENTATION = '''
+---
+module: ce_vrf
+version_added: "2.4"
+short_description: Manages VPN instance on HUAWEI CloudEngine switches.
+description:
+    - Manages VPN instance of HUAWEI CloudEngine switches.
+author: Yang yang (@CloudEngine-Ansible)
+notes:
+    - If no vrf is supplied, vrf is set to default.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vrf.py]: vtf has `required=True`, so they Playbook writer will always have to specify a vrf, so I'm not sure what this note means.

File: lib/ansible/modules/network/cloudengine/ce_vrf.py [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+      If state==absent, the route will be removed, regardless of the
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vrf.py]: `I(state=absent)`

File: lib/ansible/modules/network/cloudengine/ce_vrf.py [hunk 6]
```diff
@@ -0,0 +1,357 @@
...
+      non-required parameters.
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_vrf.py]: `non-required options.`

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by QijunPan** [comment]
@gundalow  I had fix those issues, thanks.

**Review by gundalow — APPROVED** [review]

**Pull request merged**